doc/hardware/porting/soc_porting.rst

```diff
@@ -0,0 +1,317 @@
+.. _soc_porting_guide:
+
+SoC Porting Guide
+###################
+
+To add Zephyr support for a new :term:`SoC`, you at least need a *SoC directory*
+with various files in it. Also a SoC :file:`.dtsi` in
+:zephyr_file:`dts/<ARCH>/<VENDOR>` is required.
+
+SoC Definitions
+***************
+
+It is expected that you are familiar with the board concept in Zephyr.
+A high level overview of the hardware support hierarchy and terms used in the
+Zephyr documentation can be seen in :ref:`hw_support_hierarchy`.
+
+For SoC porting, the most important terms are:
+
+- SoC: the exact system on a chip the board's CPU is part of
+- SoC series: a smaller group of tightly related SoCs
+- SoC family: a wider group of SoCs with similar characteristics
+- CPU Cluster: a cluster of one or more CPU cores.
+- CPU core: a particular CPU in an architecture
+- Architecture: an instruction set architecture
+
+Architecture
+============
+
+See :ref:`architecture_porting_guide`.
+
+
+Create your SoC directory
+*************************
+
+Each SoC must have a unique name. You need to give your SoC a unique name.
+The script ``list_hardware`` can be used to retrieve a list of all SoCs known
+in Zephyr, for example ``./scripts/list_hardware --soc-root=. --socs`` from
+Zephyr base for a list of names that are already taken, and pick something new.
+We will use ``soc1`` in the following
+
+Start by creating the directory ``zephyr/soc/<VENDOR>/soc1``, where
+``<VENDOR>`` is your vendor subdirectory.
+
+.. note::
+  A ``<VENDOR>`` subdirectory is mandatory if contributing your SoC
+  to Zephyr, but if your SoC is placed in a local repo, then any folder
+  structure under ``<your-repo>/soc`` is permitted.
+  The ``<VENDOR>`` subdirectory must match a vendor defined in the list in
+  :zephyr_file:`dts/bindings/vendor-prefixes.txt`. If the SoC vendor is not
+  having a prefix in that list, then one must be created.
```

Comment:
Is it true as well for out of tree SoC ? If yes, could it be relaxed ?

Reply:
it is already relaxed, if you read the first part:
> A ``<VENDOR>`` subdirectory is mandatory if contributing your SoC
  to Zephyr, but if your SoC is placed in a local repo, then any folder
  structure under ``<your-repo>/soc`` is permitted.

So an oot SoC can use any name, doesn't have to be ``<VENDOR>``. An oot SoC can use `foo` if it likes.
Feel free to propose a rephrase if you have a better idea.